src/ducc0/infra/threading.cc

```diff
@@ -367,22 +373,52 @@ class thread_pool
       }
   };
 
-inline thread_pool &get_pool()
+// return a pointer to a singleton ducc_thread_pool, which is always available
+inline std::shared_ptr<ducc_thread_pool> get_master_pool()
   {
-  static thread_pool pool;
+  static auto master_pool=std::make_shared<ducc_thread_pool>();
 #if __has_include(<pthread.h>)
   static std::once_flag f;
   call_once(f,
     []{
     pthread_atfork(
-      +[]{ get_pool().shutdown(); },  // prepare
-      +[]{ get_pool().restart(); },   // parent
-      +[]{ get_pool().restart(); }    // child
+      +[]{ get_master_pool()->shutdown(); },  // prepare
+      +[]{ get_master_pool()->restart(); },   // parent
+      +[]{ get_master_pool()->restart(); }    // child
       );
     });
 #endif
+  return master_pool;
+  }
 
-  return pool;
+// Simple stack of thread pools
+class PoolManager
+  {
+  private:
+    std::stack<std::shared_ptr<thread_pool>> poolstack;
```

Comment:
Instead of having a stack with `push` and `pop` it might be better to have an RAII object that manages the stack without doing any allocations.
```cpp
std::shared_ptr<thread_pool> set_pool(std::shared_ptr<thread_pool>);

class PoolGuard {
    std::shared_ptr<thread_pool> old_pool_;
public:
    PoolGuard(std::shared_ptr<thread_pool> pool) {
        old_pool_ = set_pool(pool);
    }
    ~PoolGuard() {
        set_pool(old_pool_);
    }
};
```

Reply:
Thanks, that looks great!

Reply:
Should be implemented now.